File: packages-gen/js/libs/xt-ui/frames/shell.js
const ui = require("@xtalk/ui/core.js")

function view(frame,content){
  let opts = frame["opts"];
  return ui.node("ui/column",{"class":opts["class"] || "mx-auto w-full gap-6 p-4 md:p-8"},[
    ui.node("ui/column",{"class":"gap-1"},[
      ui.node("ui/title",{"value":opts["title"] || ""},[]),
      ui.node("ui/description",{"value":opts["description"] || ""},[])
    ]),
    ui.slot(frame["id"] + "/content",content || [],{})
  ]);
}

module.exports = {["view"]:view}

File: packages-gen/dart/libs/xt-ui/lib/frames/shell.dart
import 'package:xtalk_ui/core.dart' as ui;


view(frame, content) {
  var opts = frame["opts"];
  return ui.node("ui/column",<dynamic, dynamic>{"class":opts["class"] ?? "mx-auto w-full gap-6 p-4 md:p-8"},<dynamic>[
    ui.node("ui/column",<dynamic, dynamic>{"class":"gap-1"},<dynamic>[
      ui.node(
        "ui/title",
        <dynamic, dynamic>{"value":opts["title"] ?? ""},
        <dynamic>[]
      ),
      ui.node(
        "ui/description",
        <dynamic, dynamic>{"value":opts["description"] ?? ""},
        <dynamic>[]
      )
    ]),
    ui.slot(
      frame["id"] + "/content",
      content ?? <dynamic>[],
      <dynamic, dynamic>{}
    )
  ]);
}